docs/enhancements/ova-upload.md

```diff
@@ -0,0 +1,169 @@
+---
+title: OVA Provider Storage Improvement
+authors:
+  - "@mansam"
+reviewers:
+  - "@mnecas"
+approvers:
+  - "@mnecas"
+creation-date: 2024-12-20
+last-updated: 2025-01-13
+status: implementable
+---
+
+# OVA Provider Storage Improvement
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+## Summary
+
+Allow specifying any PVC as the appliance catalogue backing for an OVA provider (rather than requiring 
+the coordinates of an NFS share) and provide a way for the end user to upload appliances to
+the catalogue.
+
+## Motivation
+
+The current implementation of the OVA provider in Forklift is difficult to use. Users
+must specify the coordinates to an NFS share that contains appliances, and adding appliances
+requires the user to add them to the NFS share outside the Forklift user experience.
+Kubernetes/OpenShift users generally expect to think of storage in terms of PVs/PVCs, so 
+having to provide an NFS share rather than a PVC (which could be backed by NFS) is
+not aligned with user expectations. Allowing the user to specify a PVC to back the OVA
+provider would be much more flexible and better match user expectations, and providing
+UI to allow appliance upload would remove friction.
+
+### Goals
+
+1. Allow specifying any PVC as the storage backing for an OVA provider, rather than requiring
+   the coordinates of an NFS share.
+2. Expose an endpoint on the OVA provider server that permits uploading appliances to the catalogue.
```

Comment:
If we will start support uploading of the OVA, we will need to create the PVC via storageclass or user pointing us to the PVC.
Is there some preexisting solution for such upload? I know Kubevirt CDI allows directly populating PVCs so maybe we could use somthing similar

Reply:
yeah sounds good to me